libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -0,0 +1,432 @@
+        self.send(request: request, completionBlock)
+        return request
+    }
+     
+    /// Execute a Composite request
+    /// - Parameter requests: Array of RestRequewst
+    /// - Parameter referenceIds: String array of reference identifiers.
+    /// - Parameter allOrNone: Boolean to indicate whether to set allOrNone.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func composite(_ requests:[RestRequest], referenceIds: [String], allOrNone: Bool, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.compositeRequest(requests, refIds: referenceIds, allOrNone: allOrNone,apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a batch request
+    /// - Parameter requests: Array of RestRequewst
+    /// - Parameter haltOnError: Boolean to indicate whether to stop or continue when error occurs in any request.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func batch(_ requests:[RestRequest], haltOnError: Bool, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.batchRequest(requests, haltOnError: haltOnError, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a SObjectTree request
+    /// - Parameter objectType: An object type.
+    /// - Parameter objectTrees: Array of SObjectTree
+    /// - Parameter completionBlock: The completion block to invoke.
+    func sObjectTree(_ objectType: String, objectTrees: [SObjectTree], _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSObjectTree: objectType, objectTrees: objectTrees, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Owned Files request
+    /// - Parameter userId: User identifier.
+    /// - Parameter page: Page number.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func ownedFiles(_ userId: String?, page: Int, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+       let request = self.request(forOwnedFilesList: userId, page: UInt(page), apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Get files in users groups
+    /// - Parameter userId: User identifier.
+    /// - Parameter page: Page number.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func filesInUsersGroups(_ userId: String?, page: Int, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
```

Comment:
Same here. I think `userId` should be required, not optional.

Reply:
Same reasons as above.